### PR TITLE
 plan: covert max/min to topN.

### LINKED ISSUE #1638: Optimize "select max(id) from t"
If there is an index on id column, we should build a better plan which will use index to calculate max(id).

### PR DESCRIPTION
**Pull request by winoros**
To make #5105 correct, we need to hold the aggregation operator. And add a `is not null` filter.
Since if there's no data, we should return `NULL` instead of empty.
If there's both not NULL data and NULLs, we should ignore the NULLs.
fix #1638

### CODE REVIEW

File: plan/aggregation_eliminate.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		if f.GetName() != ast.AggFuncMax && f.GetName() != ast.AggFuncMin {
+			return
+		}
+		// e.g. select min(-a) from t. We cannot rewrite it easily.
+		if _, ok := f.GetArgs()[0].(*expression.Column); !ok {
+			return
+		}
+
+		child := p.Children()[0]
+		// If it can be NULL, we need to filter NULL out first.
```
Comment (shenli) [inline plan/aggregation_eliminate.go]: Could you add tests covers the cases that there is null in the table and the table source result is empty.

**Comment by hanfei1991** [comment]
I think MaxMinElimination is a better name

File: plan/aggregation_eliminate.go (hunk 1 above)
Comment (winoros) [reply plan/aggregation_eliminate.go]: added

**Comment by shenli** [comment]
Please Fix CI

**Comment by shenli** [comment]
LGTM
@hanfei1991 PTAL

**Comment by ngaut** [comment]
Please fix CI

**Pull request merged**